src/Yaapii.Atoms/Map/Sorted.cs

```diff
@@ -0,0 +1,178 @@
+        /// <param name="cmp">Comparer comparing keys</param>
+        public Sorted(IEnumerable<KeyValuePair<string, Value>> pairs, IComparer<string> cmp)
+            : this(pairs, new KeyComparer<string, Value>(cmp))
+        { }
+
+        /// <summary>
+        /// Sorts the given map with the given comparer
+        /// </summary>
+        /// <param name="pairs">Map elements to be sorted</param>
+        /// <param name="cmp">Comparer comparing elements</param>
+        public Sorted(IEnumerable<KeyValuePair<string, Value>> pairs, IComparer<KeyValuePair<string, Value>> cmp)
+            : base(
+                () =>
+                {
+                    var items = new List<KeyValuePair<string, Value>>(pairs);
+                    items.Sort(cmp);
+                    return new Map.Of<string, Value>(items);
+                },
+                false
+            )
+        { }
+    }
+
+    public sealed class Sorted : Map.Envelope
+    {
+        /// <summary>
+        /// Sorts the given map with the default comperator of the key
+        /// </summary>
+        /// <param name="pairs">Map elements to be sorted</param>
+        public Sorted(IEnumerable<KeyValuePair<string, string>> pairs)
+            : this(pairs, Comparer<string>.Default)
+        { }
+
+        /// <summary>
+        /// Sorts the given map with the given compare function
+        /// </summary>
+        /// <param name="pairs">Map elements to be sorted</param>
+        /// <param name="compare">Function to compare two elements</param>
+        public Sorted(IEnumerable<KeyValuePair<string, string>> pairs, Func<KeyValuePair<string, string>, KeyValuePair<string, string>, int> compare)
+            : this(pairs, new ComparerFunc<KeyValuePair<string, string>>(compare))
+        { }
+
+        /// <summary>
+        /// Sorts the given map with the given key comparer
+        /// </summary>
+        /// <param name="pairs">Map elements to be sorted</param>
+        /// <param name="cmp">Comparer comparing keys</param>
+        public Sorted(IEnumerable<KeyValuePair<string, string>> pairs, IComparer<string> cmp)
+            : this(pairs, new KeyComparer<string, string>(cmp))
+        { }
+
+        /// <summary>
+        /// Sorts the given map with the given comparer
+        /// </summary>
+        /// <param name="pairs">Map elements to be sorted</param>
+        /// <param name="cmp">Comparer comparing elements</param>
+        public Sorted(IEnumerable<KeyValuePair<string, string>> pairs, IComparer<KeyValuePair<string, string>> cmp)
+            : base(
+                () =>
+                {
+                    var items = new List<KeyValuePair<string, string>>(pairs);
+                    items.Sort(cmp);
+                    return new Map.Of<string, string>(items);
+                },
+                false
+            )
+        { }
+    }
+
+    internal sealed class ComparerFunc<T> : IComparer<T>
```

Comment:
this should be SimpleComparer - according to our naming conventions, the type should be at the end of the name (by default)